File: BlazorApp1/Client/TypeScript/Location.ts
declare var DotNet: any;

class coordinate {
    public latitude: number = 0;
    public longitude: number = 0;
    public accuracy: number = 0;
}


let namespace = "BlazorApp1.Client";
let type = "LocationService";
async function dispatchResponse(id: string, location: coordinate) {
    await DotNet.invokeMethodAsync(namespace, 'ReceiveResponse', id, location.latitude, location.longitude, location.accuracy);
}

async function dispatchWatchResponse(id: string, location: coordinate) {
    await DotNet.invokeMethodAsync(namespace, 'ReceiveWatchResponse', id, location.latitude, location.longitude, location.accuracy);
}

window['BlazorApp1'] = window['BlazorApp1'] || {};
window['BlazorApp1']['Client'] = window['BlazorApp1']['Client'] || {};

window['BlazorApp1']['Client']['GetLocation'] = (requestId) => {
    if (navigator.geolocation) {
        navigator.geolocation.getCurrentPosition((position) => {
            dispatchResponse(requestId, position.coords);
        });
    }
    else {
        return "No location finding";
    }
};

window['BlazorApp1']['Client']['WatchLocation'] = (requestId) => {
    if (navigator.geolocation) {
        navigator.geolocation.watchPosition((position) => {
            dispatchWatchResponse(requestId, position.coords);
        });
    }
    else {
        return "No location watching";
    }
};

File: BlazorApp1/Client/Location.cs
namespace BlazorApp1.Client
{
    public class Location
    {
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double Accuracy { get; set; }
        public double Altitude { get; set; }
        public double AltitudeAccuracy { get; set; }
        public double Speed { get; set; }
        public double Heading { get; set; }

        public override string ToString()
        {
            return $"Location: ({Latitude}, {Longitude}) with accuracy {Accuracy}";
        }
    }
}
